test/support/utils.h

```diff
@@ -930,6 +932,41 @@ get_new_kernel_params(KernelParams)
 }
 #endif //TEST_DPCPP_BACKEND_PRESENT
 
+template <typename T>
+typename ::std::enable_if_t<std::is_arithmetic_v<T>, void>
+generate_arithmetic_data(T* input, std::size_t size, std::uint32_t seed)
+{
+    std::default_random_engine gen{seed};
+    std::size_t unique_threshold = 75 * size / 100;
+    if constexpr (sizeof(T) < sizeof(short)) // no uniform_int_distribution for chars
+    {
+        std::uniform_int_distribution<int> dist(std::numeric_limits<T>::lowest(), std::numeric_limits<T>::max());
+        std::generate(input, input + unique_threshold, [&] { return T(dist(gen)); });
+    }
+    else if constexpr (std::is_integral_v<T>)
+    {
+        std::uniform_int_distribution<T> dist(std::numeric_limits<T>::lowest(), std::numeric_limits<T>::max());
+        std::generate(input, input + unique_threshold, [&] { return dist(gen); });
+    }
+    else
+    {
+        // log2 - exp2 transformation allows generating floating point values,
+        // which distribution resembles uniform distribution of their bit representation
+        // This is useful for checking different cases of radix sort
+        std::uniform_real_distribution<T> dist_real(std::numeric_limits<T>::min(), log2(std::numeric_limits<T>::max()));
+        std::uniform_int_distribution<int> dist_binary(0, 1);
+        auto randomly_signed_real = [&dist_real, &dist_binary, &gen]()
+        {
+            auto v = exp2(dist_real(gen));
+            return dist_binary(gen) == 0 ? v : -v;
+        };
+        std::generate(input, input + unique_threshold, [&] { return randomly_signed_real(); });
+    }
+    for (uint32_t i = 0, j = unique_threshold; j < size; ++i, ++j)
```

Comment:
It makes sense to add an assert before the loop, like:
`assert(unique_threshold >= size/2 && unique_threshold < size);`

Reply:
Done